blocks/api/registration.js

```diff
@@ -24,6 +24,17 @@ const blocks = {};
 
 const categories = getCategories();
 
+/**
+ * Current user properties.
+ *
+ * @type {Object}
+ */
+const currentUser = get(
+	Object.values( pickBy( window._wpAPIDataPreload, ( value, key ) => key.indexOf( '/wp/v2/users/me' ) !== -1 ) ),
```

Comment:
> You might also imagine an existing post which had been authored by another user with the capability. If we don't allow the block to be registered, it may appear as broken.

Good Point. In the classic editor this can only happen if the upper roles (author, editor, administrator), keep the post as pending. As soon as the post is published, the contributor cannot edit it anymore.